images/virtualization-artifact/pkg/controller/vmclass/internal/validators/sizing_policies_validator.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validators
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/deckhouse/virtualization/api/core/v1alpha2"
+)
+
+type SizingPoliciesValidator struct {
+	client client.Client
+}
+
+func NewSizingPoliciesValidator(client client.Client) *SizingPoliciesValidator {
+	return &SizingPoliciesValidator{client: client}
+}
+
+func (v *SizingPoliciesValidator) ValidateCreate(_ context.Context, vmclass *v1alpha2.VirtualMachineClass) (admission.Warnings, error) {
+	if HasCpuSizePoliciesCrosses(&vmclass.Spec) {
+		return nil, fmt.Errorf("vmclass %s has size policy cpu crosses", vmclass.Name)
+	}
+
+	return nil, nil
+}
+
+func (v *SizingPoliciesValidator) ValidateUpdate(_ context.Context, _, newVMClass *v1alpha2.VirtualMachineClass) (admission.Warnings, error) {
+	if HasCpuSizePoliciesCrosses(&newVMClass.Spec) {
+		return nil, fmt.Errorf("vmclass %s has size policy cpu crosses", newVMClass.Name)
+	}
+
+	return nil, nil
+}
+
+func HasCpuSizePoliciesCrosses(vmclass *v1alpha2.VirtualMachineClassSpec) bool {
+	usedPairs := make(map[[2]int]bool)
```

Comment:
```suggestion
	usedPairs := make(map[[2]int]struct{})
```

you don't use value. Could you replace value to struct